lib/internal/queue.js

```diff
@@ -45,16 +47,14 @@ export default function queue(worker, concurrency, payload) {
     }
 
     function _next(tasks) {
-        return function(){
+        return rest(function(args){
             workers -= 1;
 
-            var removed = false;
-            var args = arguments;
             arrayEach(tasks, function (task) {
-                arrayEach(workersList, function (worker, index) {
-                    if (worker === task && !removed) {
+                arraySome(workersList, function (worker, index) {
```

Comment:
The thing is we don't remove them on each run because we check if `removed
= true` so we only remove one task at a time while possibly running multiple

On Tue, Jul 5, 2016 at 2:32 AM, Alex Early <notifications@github.com> wrote:

> In lib/internal/queue.js
> <https://github.com/caolan/async/pull/1214#discussion_r69512608>:
>
> >              arrayEach(tasks, function (task) {
> > -                arrayEach(workersList, function (worker, index) {
> > -                    if (worker === task && !removed) {
> > +                arraySome(workersList, function (worker, index) {
>
> I'm pretty sure we only execute each task once, because we remove them as
> they are processed. I could be missing some detail though.
>
> I think using some for short circuiting is a bit hacky. It also means we
> don't have to import arraySome.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/caolan/async/pull/1214/files/ef05d55e1bbe1a8c823eaa4d5e0de6f5818ff27e#r69512608>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADUIEMOBWBFDg4bzlY00kLOVwOpbCUKfks5qSfp2gaJpZM4JD-Av>
> .
>